File: Models/Expense.cs
using System.Text.Json.Serialization;

namespace LedgerFlow.API.Models 
{
    public class Expense
    {
        public int Id {get; set; }
        public int UserId {get; set; }
        public decimal Amount {get; set; }
        public string Description {get; set; }
        public string Category {get; set; }
        public int? CategoryId { get; set; }
        public string Status {get; set; } // Pending, Approved, Rejected, Reimbursed
        public DateTime SubmittedAt { get; set; }
        public DateTime? ApprovedAt { get; set; }
        public DateTime? ReimbursedAt { get; set; }
        public string? RejectionReason { get; set; }

        /// <summary>Tenant isolation.</summary>
        public int CompanyId { get; set; }

        [JsonIgnore]
        public User? User {get; set; }

        public Category? CategoryRef { get; set; }

        [JsonIgnore]
        public ICollection<Receipt>? Receipts { get; set; }
    }
}

File: ledgerflow-frontend/src/types/expense.ts
/**
 * Expense Types - Complete Definition
 */

export type ExpenseStatus = 'Pending' | 'Approved' | 'Rejected' | 'Reimbursed';
export type ExpenseCategory = 'Travel' | 'Food' | 'Office' | 'Equipment' | 'Other';

export interface Expense {
  id: number;
  userId: number;
  userName: string;
  companyId: number;
  amount: number;
  category: string;
  description: string;
  receiptUrl?: string;
  receipts?: Receipt[]; // Array of receipt attachments
  status: ExpenseStatus;
  submittedAt: string;
  approvedAt?: string;
  rejectedAt?: string;
  reimbursedAt?: string;
  approvedBy?: number;
  approverName?: string;
  rejectionReason?: string;
  createdAt?: string; // Alternative to submittedAt
}

export interface Receipt {
  id?: number;
  fileUrl: string;
  fileName: string;
  uploadedAt?: string;
}

export interface ApprovalHistory {
  id: number;
  expenseId: number;
  expenseDescription?: string; // Description of the expense
  expenseAmount?: number; // Amount of the expense
  approverId: number;
  approverName: string;
  approverEmail?: string;
  action: 'Approved' | 'Rejected';
  status?: 'Approved' | 'Rejected'; // Alias for action
  comments?: string;
  remarks?: string; // Alternative to comments
  timestamp: string;
  createdAt?: string; // Alternative to timestamp
}

export interface CreateExpenseRequest {
  amount: number;
  category: string;
  description: string;
  receiptUrl?: string;
}

export interface ApproveExpenseRequest {
  expenseId: number;
  comments?: string;
}

export interface RejectExpenseRequest {
  expenseId: number;
  reason: string;
}

export interface ExpenseStats {
  totalExpenses: number;
  pendingExpenses: number;
  approvedExpenses: number;
  rejectedExpenses: number;
  reimbursedExpenses: number;
  totalAmount: number;
}
